src/project_types.rs

```diff
@@ -15,12 +22,32 @@ pub struct User {
     pub public_key: String,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, RustcEncodable, RustcDecodable)]
+pub struct SequenceSection {
+    pub seq_name: String,
+    pub index: u32,
+    pub num_frames: u32,
+    pub editor: Option<User>,
+    pub data: Vec<Vec<u8>>, // Row is channel, column is frame    
```

Comment:
I'm kind of worried about using Vec<Vec<>> here. Vectors are a pointer to a heap allocated memory chunk. So a Vec<Vec> is basically an array of pointers to other arrays, which could be of different length. I looked at ndarray a bit, but I don't think its a problem for now. But when writing the music playing code, we might change this out.